src/components/customers/CustomerPaymentMethods.tsx

```diff
@@ -0,0 +1,215 @@
+import { gql } from '@apollo/client'
+import { useEffect, useRef, useState } from 'react'
+
+import { LinkedPaymentProvider } from '~/components/customers/types'
+import {
+  Alert,
+  Button,
+  Dialog,
+  DialogRef,
+  Skeleton,
+  Tooltip,
+  Typography,
+} from '~/components/designSystem'
+import { ComboBox } from '~/components/form'
+import { PageSectionTitle } from '~/components/layouts/Section'
+import { addToast } from '~/core/apolloClient'
+import { copyToClipboard } from '~/core/utils/copyToClipboard'
+import {
+  CustomerMainInfosFragment,
+  ProviderPaymentMethodsEnum,
+  useGenerateCheckoutUrlMutation,
+} from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+
+export const ADD_PAYMENT_METHOD_TEST_ID = 'add-payment-method-dialog'
+export const ELIGIBLE_PAYMENT_METHODS_TEST_ID = 'eligible-payment-methods-text'
+export const INELIGIBLE_PAYMENT_METHODS_TEST_ID = 'ineligible-payment-methods-text'
+export const EMPTY_STATE_TEST_ID = 'no-payment-methods-available-text'
+export const GENERATE_CHECKOUT_URL_BUTTON_TEST_ID = 'generate-checkout-url-button'
+export const CANCEL_DIALOG_BUTTON_TEST_ID = 'cancel-dialog-button'
+export const ERROR_ALERT_TEST_ID = 'error-alert'
+export const CHECKOUT_URL_TEXT_TEST_ID = 'checkout-url-text'
+
+interface Props {
+  customer: CustomerMainInfosFragment
+  linkedPaymentProvider: LinkedPaymentProvider
+}
+
+gql`
+  mutation generateCheckoutUrl($input: GenerateCheckoutUrlInput!) {
+    generateCheckoutUrl(input: $input) {
+      checkoutUrl
+    }
+  }
+`
+
+export const CustomerPaymentMethods = ({ customer, linkedPaymentProvider }: Props) => {
+  const { translate } = useInternationalization()
+  const [hasError, setHasError] = useState(false)
+  const [generateCheckoutUrlMutation, { data, loading, error }] = useGenerateCheckoutUrlMutation({
+    variables: {
+      input: { customerId: customer.id },
+    },
+  })
+
+  const addPaymentDialogRef = useRef<DialogRef>(null)
+
+  const [selectedPaymentProvider, setSelectedPaymentProvider] = useState<string>('')
+
+  const customerAvailablePaymentMethods = customer?.providerCustomer?.providerPaymentMethods || []
+  const hasAtLeastOneAvailablePaymentMethod = customerAvailablePaymentMethods.length > 0
+
+  const ineligiblePaymentMethods = [
+    ProviderPaymentMethodsEnum.CustomerBalance,
+    ProviderPaymentMethodsEnum.Crypto,
+  ]
+
+  const isCustomerEligibleForAddingPaymentMethods = customerAvailablePaymentMethods.some(
+    (method) => !ineligiblePaymentMethods.includes(method),
+  )
+
+  const hasMissingConnectedPaymentProvider =
+    hasAtLeastOneAvailablePaymentMethod && !isCustomerEligibleForAddingPaymentMethods
+
+  const mapOptions = linkedPaymentProvider ? [linkedPaymentProvider] : []
+
+  const connectedPaymentProvidersData = mapOptions.map(({ code, name }) => ({
+    value: code,
+    label: name,
+  }))
+
+  const hasOneAvailableOption = connectedPaymentProvidersData.length === 1
```

Comment:
May no need change, just trying to understand better the code.
If I understand correctly this condition will always be true, correct?
I mean, the length will either be 1 or 0, and (for now) cannot be more than that today?

Reply:
Guess it will help the day we can have more than 1, ok thanks!